truffle/com.oracle.truffle.tck/src/com/oracle/truffle/tck/TruffleTCK.java

```diff
@@ -995,33 +1023,38 @@ public void testPrimitiveIdentityForeignObject() throws Exception {
     @Test
     public void testCoExistanceOfMultipleLanguageInstances() throws Exception {
         final String countMethod = countInvocations();
-        PolyglotEngine.Value count1 = findGlobalSymbol(countMethod);
-        PolyglotEngine vm1 = tckVM;
-        tckVM = null; // clean-up
-        PolyglotEngine.Value count2 = findGlobalSymbol(countMethod);
-        PolyglotEngine vm2 = tckVM;
-
-        assertNotSame("Two virtual machines allocated", vm1, vm2);
-
-        int prev1 = 0;
-        int prev2 = 0;
-        StringBuilder log = new StringBuilder();
-        for (int i = 0; i < 10; i++) {
-            int quantum = RANDOM.nextInt(10);
-            log.append("quantum" + i + " is " + quantum + "\n");
-            for (int j = 0; j < quantum; j++) {
-                Object res = count1.execute().get();
-                assert res instanceof Number : "expecting number: " + res + "\n" + log;
-                ++prev1;
-                assert ((Number) res).intValue() == prev1 : "expecting " + prev1 + " but was " + res + "\n" + log;
-            }
-            for (int j = 0; j < quantum; j++) {
-                Object res = count2.execute().get();
-                assert res instanceof Number : "expecting number: " + res + "\n" + log;
-                ++prev2;
-                assert ((Number) res).intValue() == prev2 : "expecting " + prev2 + " but was " + res + "\n" + log;
+        PolyglotEngine.Builder builder = PolyglotEngine.newBuilder().executor(Executors.newSingleThreadExecutor());
+        PolyglotEngine vm2 = prepareVM(builder);
+        try {
+            PolyglotEngine.Value count2 = vm2.findGlobalSymbol(countMethod);
+
+            PolyglotEngine.Value count1 = findGlobalSymbol(countMethod);
+            PolyglotEngine vm1 = tckVM;
```

Comment:
How does it work here, since it seems both `vm1` and `vm2` are used from the same thread?